File: apps/problems/search-a-2d-matrix/boilerplate-full/function.cpp
#include <iostream>
#include <vector>
#include <string>
#include <sstream>
#include <climits>



//##USER_CODE_HERE


int main() {
    int m_matrix;
std::cin >> m_matrix;
int n_matrix;
std::cin >> n_matrix;
std::vector<std::vector<int>> matrix(m_matrix, std::vector<int>(n_matrix));

for(int i = 0; i < m_matrix; ++i) {
    for(int j = 0; j < n_matrix; ++j) {
        std::cin >> matrix[i][j];
    }
}
int target;
std::cin >> target;
    Solution obj;
    bool result = obj.searchMatrix(matrix, target);
    std::cout << (result ? "true" : "false") << std::endl;
    return 0;
}

File: apps/problems/search-a-2d-matrix/boilerplate-full/function.java

    import java.io.*;
    import java.util.*;

    

    //##USER_CODE_HERE##

    public class Main {
      
    public static void main(String[] args) throws IOException {
      BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
      int rows_matrix = Integer.parseInt(br.readLine().trim());
        int cols_matrix = Integer.parseInt(br.readLine().trim());
        int[][] matrix = new int[rows_matrix][cols_matrix];
        for (int i = 0; i < rows_matrix; i++) {
          String[] row = br.readLine().trim().split("\\s+");
          for (int j = 0; j < cols_matrix; j++) {
            matrix[i][j] = Integer.parseInt(row[j]);
          }
        }
int target = Integer.parseInt(br.readLine().trim());
      Solution obj = new Solution();
      boolean result = obj.searchMatrix(matrix, target);
      System.out.println(result);
      br.close();
    }
      
    }
  